src/pages/home/report/ReportActionCompose.js

```diff
@@ -328,32 +330,34 @@ class ReportActionCompose extends React.PureComponent {
             selection: this.selection,
         });
         this.selection = newSelection;
+        this.nextSelectionAfterEmojiInsertion = newSelection;
         this.updateComment(newText, true);
     }
 
     /**
-     * Focus the composer text input
-     * @param {Boolean} [shouldelay=false] Impose delay before focusing the composer
-     * @memberof ReportActionCompose
+     * This will be called when the emoji picker modal closes.
+     * Once that's closed we want to focus the text input again and
+     * set the selection to the new position if an emoji was added.
      */
-    focus(shouldelay = false) {
-        // There could be other animations running while we trigger manual focus.
-        // This prevents focus from making those animations janky.
-        InteractionManager.runAfterInteractions(() => {
-            if (!this.textInput) {
-                return;
-            }
-
-            if (!shouldelay) {
-                this.textInput.focus();
-            } else {
-                // Keyboard is not opened after Emoji Picker is closed
-                // SetTimeout is used as a workaround
-                // https://github.com/react-native-modal/react-native-modal/issues/114
-                // We carefully choose a delay. 100ms is found enough for keyboard to open.
-                setTimeout(() => this.textInput.focus(), 100);
-            }
-        });
+    focusInputAndSetSelection() {
+        // We first need to focus the input, and then set the selection, as otherwise
+        // the focus might cause the selection to be set to the end of the text input
+        this.textInput.focus(
+            () => {
+                if (!this.nextSelectionAfterEmojiInsertion) {
+                    return;
+                }
+
+                requestAnimationFrame(() => {
+                    this.selection = this.nextSelectionAfterEmojiInsertion;
+                    this.textInput.setSelection(this.selection.start, this.selection.end);
+                    this.nextSelectionAfterEmojiInsertion = null;
+                });
+            },
+
+            // run the focus with a delay. Note: Its platform dependent whether there should be a delay or not.
+            true,
```

Comment:
```suggestion
            // Run the focus with a delay. Note: Its platform dependent whether there should be a delay or not.
            true,
```

I do not understand the comment. Both index.js and index.native.js files are using `delay`.